opcua/gopcua/subscribe.go

```diff
@@ -99,15 +108,15 @@ func (c client) Subscribe(cfg opcua.Config) error {
 	}
 	defer sub.Cancel()
 
-	if err := c.runHandler(sub, cfg); err != nil {
-		return err
+	if err := c.runHandler(sub, cfg.ServerURI, cfg.NodeID); err != nil {
+		c.logger.Warn(fmt.Sprintf("Unsubscribed from OPC-UA node %s.%s: %s", cfg.ServerURI, cfg.NodeID, err))
 	}
 
 	return nil
 }
 
-func (c client) runHandler(sub *opcuaGopcua.Subscription, cfg opcua.Config) error {
-	nodeID, err := uaGopcua.ParseNodeID(cfg.NodeID)
+func (c client) runHandler(sub *opcuaGopcua.Subscription, uri, node string) error {
```

Comment:
this method maybe should be split in smaller methods

Reply:
I agree, it's just that for now I'm trying to keep this code similar to an example that I'm using

Reply:
so you call it here https://github.com/manuio/mainflux/blob/opcua/opcua/gopcua/subscribe.go#L111
and it stays in runHandler until error occurs?

Reply:
yes